core/src/main/java/bisq/core/account/sign/SignedWitness.java

```diff
@@ -0,0 +1,156 @@
+        DateTolerantPayload, CapabilityRequiringPayload {
+    private static final long TOLERANCE = TimeUnit.DAYS.toMillis(1);
+
+    private final boolean signedByArbitrator;
+    private final byte[] witnessHash;
+    private final byte[] signature;
+    private final byte[] signerPubKey;
+    private final byte[] witnessOwnerPubKey;
+    private final long date;
+    //TODO should we add trade amount?
+
+    transient private final byte[] hash;
+
+    public SignedWitness(boolean signedByArbitrator,
+                         byte[] witnessHash,
+                         byte[] signature,
+                         byte[] signerPubKey,
+                         byte[] witnessOwnerPubKey,
+                         long date) {
+        this.signedByArbitrator = signedByArbitrator;
+        this.witnessHash = witnessHash;
+        this.signature = signature;
+        this.signerPubKey = signerPubKey;
+        this.witnessOwnerPubKey = witnessOwnerPubKey;
+        this.date = date;
+
+        byte[] data = Utilities.concatenateByteArrays(witnessHash, signature);
+        data = Utilities.concatenateByteArrays(data, signerPubKey);
+        hash = Hash.getSha256Ripemd160hash(data);
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // PROTO BUFFER
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public PB.PersistableNetworkPayload toProtoMessage() {
+        final PB.SignedWitness.Builder builder = PB.SignedWitness.newBuilder()
+                .setSignedByArbitrator(signedByArbitrator)
+                .setWitnessHash(ByteString.copyFrom(witnessHash))
+                .setSignature(ByteString.copyFrom(signature))
+                .setSignerPubKey(ByteString.copyFrom(signerPubKey))
+                .setWitnessOwnerPubKey(ByteString.copyFrom(witnessOwnerPubKey))
+                .setDate(date);
+        return PB.PersistableNetworkPayload.newBuilder().setSignedWitness(builder).build();
+    }
+
+    public PB.SignedWitness toProtoSignedWitness() {
+        return toProtoMessage().getSignedWitness();
+    }
+
+    public static SignedWitness fromProto(PB.SignedWitness proto) {
+        return new SignedWitness(proto.getSignedByArbitrator(),
+                proto.getWitnessHash().toByteArray(),
+                proto.getSignature().toByteArray(),
+                proto.getSignerPubKey().toByteArray(),
+                proto.getWitnessOwnerPubKey().toByteArray(),
+                proto.getDate());
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // API
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public boolean isDateInTolerance() {
+        // We don't allow older or newer then 1 day.
+        // Preventing forward dating is also important to protect against a sophisticated attack
```

Comment:
The date is not part of the hash. Is this by intent?

Reply:
Will think about it

Reply:
Just added a comment:
  // Date is not added to hash to avoid that the same account with same sigs can be stored multiple 
        // times if date would differ